addons/mail/static/src/models/thread/thread.js

```diff
@@ -495,63 +540,65 @@ function factory(dependencies) {
         }
 
         /**
-         * Open provided thread, either in discuss app or as a chat window.
+         * Opens this thread, either as form view, in Discuss app or as a
+         * chat window.
          *
-         * @param {Object} param0
-         * @param {string} [param0.chatWindowMode='last_visible']
-         * @param {boolean} [param0.resetDiscussDomain=false]
+         * @param {Object} [param0]
+         * @param {boolean} [param0.chatWindowReplaceNewMessage=false]
+         * @param {boolean} [param0.makeActive=true]
+         * @param {boolean} [param0.openExpanded=false]
          */
-        open({ chatWindowMode = 'last_visible', resetDiscussDomain = false } = {}) {
+        async open({ chatWindowReplaceNewMessage = false, makeActive = true, openExpanded = false } = {}) {
             const device = this.env.messaging.device;
             const discuss = this.env.messaging.discuss;
-            const messagingMenu = this.env.messaging.messagingMenu;
+
+            // Can thread only be opened in form view?
             if (!['mail.box', 'mail.channel'].includes(this.model)) {
-                this.env.messaging.openDocument({
-                    id: this.id,
-                    model: this.model,
-                });
+                // Opening form view is always disruptive, so it only applies
+                // if making active is requested.
+                if (makeActive) {
+                    return this.env.messaging.openDocument({
+                        id: this.id,
+                        model: this.model,
+                    });
+                }
+                return;
             }
+
+            // Should thread be opened in discuss?
             if (
-                (!device.isMobile && discuss.isOpen) ||
-                (device.isMobile && this.model === 'mail.box')
+                (!device.isMobile && (discuss.isOpen || openExpanded)) ||
+                this.model === 'mail.box'
             ) {
-                if (resetDiscussDomain) {
-                    discuss.threadViewer.update({ stringifiedDomain: '[]' });
+                // Opening in Discuss is always disruptive, so it only applies
+                // if making active is requested.
+                if (makeActive) {
+                    return discuss.openThread(this);
```

Comment:
it shows how stupid the API is with discuss. The notion of making a thread active should be limited to chat windows...